File: sololearn/tasks/anagram_check.js
function isAnagram(a, b) {
    if (a.length !== b.length) return false;
    const count = {};
    for (const c of a) count[c] = (count[c] || 0) + 1;
    for (const c of b) {
        if (!count[c]) return false;
        count[c]--;
    }
    return true;
}

File: sololearn/lua_snippets/anagram_check.lua
-- Task: Determine if two strings are anagrams
-- Uses hash tables for counting

function areAnagrams(a, b)
  if #a ~= #b then return false end

  local count = {}
  for i = 1, #a do
    local c = a:sub(i, i)
    count[c] = (count[c] or 0) + 1
  end

  for i = 1, #b do
    local c = b:sub(i, i)
    if not count[c] or count[c] == 0 then
      return false
    end
    count[c] = count[c] - 1
  end

  return true
end
